### DISCUSSION TITLE
Does micropython need to include the PSF license in its own license?

### DISCUSSION THREAD

**Discussion by edy14431** [micropython/micropython#12076] [General]
Is micropython a derivative version or a derivative work of python?

The Micropython python README.md file says the following:
"This is the MicroPython project, which aims to put an implementation of Python 3.x on microcontrollers and small embedded systems. You can find the official website at micropython.org. [...] MicroPython implements the entire Python 3.4 syntax (including exceptions, with, yield from, etc., and additionally async/await keywords from Python 3.5 and some select features from later versions)"

Now see what this part of the PSF license I found in the path lib>micropython-lib says about derivative versions and derivative works: 
"2. Subject to the terms and conditions of this License Agreement, PSF hereby
grants Licensee a nonexclusive, royalty-free, world-wide license to reproduce,
analyze, test, perform and/or display publicly, prepare derivative works,
distribute, and otherwise use Python alone or in any derivative version,
provided, however, that PSF's License Agreement and PSF's notice of copyright,
i.e., "Copyright (c) 2001, 2002, 2003, 2004, 2005, 2006, 2007, 2008, 2009, 2010,
2011, 2012, 2013 Python Software Foundation; All Rights Reserved" are retained
in Python alone or in any derivative version prepared by Licensee."

If Micropython is a derivative version or a derivative work of Python, wouldn't it be necessary for the Micropython project to provide in its own license one or a few PSF licenses that cover the Python versions used in Micropython, such as versions 3.4 and 3.5?

**Comment by mattytrentini** [comment]
MicroPython is a _reimplementation_ of Python, the language. Not derived from CPython code.

**Comment by jimmo** [comment]
@edy14431 MicroPython (and everything in the main repo) itself is not derived from the CPython source code, it is an entirely from-scratch implementation of the Python language and standard library. Python's license applies to "software in source or binary form and its associated documentation", and MicroPython is derived from none of those things.

However, there are some packages in micropython-lib that are derived in some form from CPython versions of the same library, which is why the micropython-lib LICENSE also includes the Python license. None of these packages are included in any released firmware or official builds from micropython.org.

The missing detail here is that micropython-lib is now a submodule of the main repo, and therefore should also have a corresponding entry in the main repo's LICENSE where we list all the submodules and third-party code (which should indicate that lib/micropython-lib contains both MIT and Python-licensed code).

(Of course, usual disclaimer that you should consult legal advice if this is important for your project or product).
